src/interfaces/swig/SGBase.i

```diff
@@ -237,6 +237,29 @@ public void readExternal(java.io.ObjectInput in) throws java.io.IOException, jav
         }
 %}
 
+// taken from https://github.com/swig/swig/issues/723#issuecomment-230178855
+%typemap(out) void _swig_monkey_patch "$result = PyErr_Occurred() ? NULL : SWIG_Py_Void();"
+%inline %{
+    static void _swig_monkey_patch(PyObject *type, PyObject *name, PyObject *object) {
+        if (PyType_Check(type)) {
+#if PY_VERSION_HEX>=0x03000000
```

Comment:
omg how lovely :D

Reply:
what is the perspective on this bit here?

Reply:
I think for this purpose it is fine, but in general I don't think it is advisable to change the behaviour of software at runtime... This just seems like the simplest solution without messing with the C++ code.